### DISCUSSION TITLE
Using ww3_uprstr to update the unstructured restart files

### DISCUSSION THREAD

**Discussion by Liu-Jincan** [NOAA-EMC/WW3#883] [Q&A]
Is it possible to update restart files using **ww3_uprstr** when the grid is **unstructured**? I have seen only two grbtxt format in the [ww3_uprstr.F90](https://github.com/NOAA-EMC/WW3/blob/develop/model/src/ww3_uprstr.F90), and I think they are not suitable for the unstructured restart files. Any suggestion is helpful. Thanks.

**Comment by JessicaMeixner-NOAA** [comment]
Great question! I don't believe ww3_upstr has ever been tested for unstructured grids, but I suspect it could be updated to use them with a few modifications.

**Reply by Liu-Jincan** [reply]
Thanks. I am not familiar with the variables in the [ww3_uprstr.F90](https://github.com/NOAA-EMC/WW3/blob/develop/model/src/ww3_uprstr.F90), but I wll have a try.

**Comment by flampouris** [comment]
The data in the grbtxt is a column following the order of the restart file. 

As long as the grid info from the restart file of the background is used to export your analysis, the uprstrt.ftn subroutine can be used as is. 

--- 

If you permit one question, do you run DA on an unstructured grid?

**Reply by Liu-Jincan** [reply]
Yes. I really want to assimilate the observations into the unstructured grid wave model.

**Reply by sunleqiang** [reply]
> The data in the grbtxt is a column following the order of the restart file.
> 
> As long as the grid info from the restart file of the background is used to export your analysis, the uprstrt.ftn subroutine can be used as is.
> 
> If you permit one question, do you run DA on an unstructured grid?

I think the real challenge is not in ww3_uprstr, but in the DA algorithm. For example, the calculation of the distance between observation and target analysis grid point might be tricky in unstructured grid.

**Reply by aronroland** [reply]
There is no assimilation in ww3 for unstructured grids ...

**Reply by flampouris** [reply]
There is no assimilation in ww3.-

**Reply by aliabdolali** [reply]
Sofar Ocean has developed DA for ww3, and we were envisioning expanding it to our applications
See this paper
https://doi.org/10.1029/2022GL098973

**Comment by aronroland** [comment]
great!--Sent from my Android phone with GMX Mail. Please excuse my brevity.On 4/4/23, 2:22 AM "Ali.Abdolali" ***@***.***> wrote:

   
  Sofar Ocean has developed DA for ww3, and we were envisioning expanding it to our applications See this paper https://doi.org/10.1029/2022GL098973
